app/src/main/java/com/example/eric/memberquiz/GameActivity.java

```diff
@@ -0,0 +1,306 @@
+                    editor.putInt("score", prefScore); // Storing the new user score
+                    editor.apply(); //saving the changes
+                }
+                timer.cancel();
+                finish();
+                startActivity(getIntent());
+            }
+        });
+
+        button4.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String buttonText = button4.getText().toString();
+                if (!buttonText.equalsIgnoreCase(contactName)){
+                    Toast.makeText(getApplicationContext(), "Wrong!", Toast.LENGTH_SHORT).show();
+                } else {
+                    // SharedPrefs Object to save score
+                    int prefScore = pref.getInt("score", 0) + 1; // getting last score from the SharedPrefs, if there is no previous score it sets it to 0
+                    SharedPreferences.Editor editor = pref.edit();
+                    editor.putInt("score", prefScore); // Storing the new user score
+                    editor.apply(); //saving the changes
+                }
+                timer.cancel();
+                finish();
+                startActivity(getIntent());
+            }
+        });
+
+        endButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                AlertDialog alertDialog = new AlertDialog.Builder(GameActivity.this).create();
+                alertDialog.setTitle("End Game");
+                alertDialog.setMessage("Are you sure you want to quit?");
+                alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "NO",
+                        new DialogInterface.OnClickListener(){
+                            public void onClick(DialogInterface dialog, int which){
+                                dialog.dismiss();
+                            }
+                        });
+                alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "YES",
+                        new DialogInterface.OnClickListener() {
+                            public void onClick(DialogInterface dialog, int which) {
+                                SharedPreferences.Editor editor = pref.edit();
+                                editor.putInt("score", 0);
+                                editor.apply();
+                                finish();
+                                timer.cancel();
+                                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+                                startActivity(intent);
+
+                            }
+                        });
+                alertDialog.show();
+
+            }
+        });
+    }
+
+    @Override
+    public void onStop(){
+        timer.cancel();
+        super.onStop();
+    }
+
+    @Override
+    public void onRestart(){
+        super.onRestart();
+        timer2 = new CountDownTimer(5500, 1000) {
```

Comment:
Again, if you're going to repeat a block of code just copy paste it into a private method.